src/main/java/org/broadinstitute/hellbender/tools/walkers/fasta/FastaReferenceMaker.java

```diff
@@ -115,7 +119,12 @@ protected void advancePosition(SimpleInterval interval) {
     private void finalizeSequence() {
         final String description = lastPosition.getContig() + ":" + currentSequenceStartPosition + "-" + lastPosition.getEnd();
         try {
-            writer.appendSequence(String.valueOf(contigCount), description, basesPerLine, Bytes.toArray(sequence));
+            if(keepContigNames) {
+                writer.appendSequence(lastPosition.getContig(), description, basesPerLine, Bytes.toArray(sequence));
+            }
+            else {
+                writer.appendSequence(String.valueOf(contigCount), description, basesPerLine, Bytes.toArray(sequence));
```

Comment:
If you're using just a slice of a contig, perhaps the name should include the full genomic coordinates